packages/eslint-plugin/src/rules/no-other-module-relative-import.ts

```diff
@@ -0,0 +1,65 @@
+import { Rule } from "eslint";
+import path from "path";
+
+function parentDirCount(dir: string) {
+    const match = dir.match(/^(\.\.\/)*/);
+    return match[0].length / 3;
+}
+
+export default {
+    meta: {
+        type: "suggestion",
+        fixable: "code",
+        schema: [
+            {
+                type: "object",
+                properties: {
+                    sourceRoot: {
+                        type: "string",
+                    },
+                    sourceRootAlias: {
+                        type: "string",
+                    },
+                },
+                additionalProperties: false,
+            },
+        ],
+    },
+    create(context) {
+        return {
+            ImportDeclaration: function (node) {
+                const options = context.options[0] ?? { sourceRoot: "./src", sourceRootAlias: "@src" };
+
+                const importParentDirCount = parentDirCount(node.source.value.toString());
+                if (!importParentDirCount) return;
```

Comment:
IMO this should be after the filePath check, maybe even after the relativeFileToSourceDir check

Reply:
I added a comment but did not change it, the loactions seems perfect for me, right after we know the import path is not relative we return early, as this case is irellevant for this rule.